renaissance-core/src/main/java/org/renaissance/Benchmark.java

```diff
@@ -27,64 +39,185 @@
-
+  @interface Repetitions {
     int value();
+  }
+
+
+  /**
+   * Container for {@link Configuration} annotations. Does not have to be
+   * used explicitly since Java 8 which introduced the {@link Repeatable}
+   * annotation. However, older Scala versions do not support annotations
+   * types tagged as {@link Repeatable}, therefore the container type has
+   * to be used explicitly.
+   */
+  @Documented
+  @Target(ElementType.TYPE)
+  @Retention(RetentionPolicy.RUNTIME)
+  @interface Configurations {
+    Configuration[] value() default {};
+  }
+
 
+  /**
+   * Defines a named benchmark configuration that a benchmark can be run with.
+   * <p>
+   * Each configuration must have a {@link #name() name} and consists of the
+   * parameters declared using the {@link Parameter} annotation, initially set
+   * to their default values.
+   * <p>
+   * The {@link #settings() settings} attribute is optional and allows overriding
+   * the values of declared configurable parameters (if not set, the parameters
+   * are left at their default values). The attribute is an array of strings,
+   * each containing a single key-value pair assignment in the form
+   * {@code "<parameter_name> = <parameter_value>"}, which overrides the value
+   * of the given parameter. Only parameters that have been previously
+   * declared using the {@link Parameter} annotation can be overriden.
+   * <p>
+   * The {@link #summary() summary} description is optional but recommended.
+   */
+  @Documented
+  @Target(ElementType.TYPE)
+  @Repeatable(Configurations.class)
+  @Retention(RetentionPolicy.RUNTIME)
+  @interface Configuration {
+    String name();
+    String[] settings() default {};
+    String summary() default "";
   }
 
+
+  /**
+   * Container for {@link Parameter} annotations. Does not have to be
+   * used explicitly since Java 8 which introduced the {@link Repeatable}
+   * annotation. However, older Scala versions do not support annotations
+   * types tagged as {@link Repeatable}, therefore the container type has
+   * to be used explicitly.
+   */
+  @Documented
+  @Target(ElementType.TYPE)
+  @Retention(RetentionPolicy.RUNTIME)
+  @interface Parameters {
+    Parameter[] value() default {};
+  }
+
+
+  /**
+   * Defines a configurable benchmark parameter that a benchmark can query.
+   * <p>
+   * Each parameter must have a name and a default value represented as
+   * {@link String}. A summary description of the parameter is optional but
+   * recommended. The default values of configurable parameters associated
+   * with a benchmark class make up a default configuration (the configuration
+   * is named {@code default} and does not need to be declared explicitly).
+   */
+  @Documented
+  @Target(ElementType.TYPE)
+  @Repeatable(Parameters.class)
+  @Retention(RetentionPolicy.RUNTIME)
+  @interface Parameter {
+    String name();
+    String defaultValue() default "";
+    String summary() default "";
+  }
+
+  //
```

Comment:
left over?

Reply:
This one is intentional (and all the others except the one with commented out line) -- I use empty EOL comments to separate blocks of related methods. I can stop doing that if people object.